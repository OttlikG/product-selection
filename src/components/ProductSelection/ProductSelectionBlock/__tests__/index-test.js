
import React from 'react'
import ProductSelectionBox from '../index'

jest.mock('../Basket', () => 'Basket')
jest.mock('../News', () => 'News')
jest.mock('../Sports', () => 'Sports')

describe('ProductSelectionBox', () => {
  let props, component
  const chelsea = {
    category: 'Sports',
    id: 'aaa',
    location: 'LONDON',
    product_name: 'Chelsea TV',
    isSelected: true
  }
  const arsenal = {
    category: 'Sports',
    id: 'bbb',
    location: 'LONDON',
    product_name: 'Arsenal TV',
    isSelected: false
  }

  beforeEach(() => {
    props = {
      sports: [chelsea, arsenal],
      news: [],
      selectedProducts: [chelsea],
      selectProductById: jest.fn(),
      push: jest.fn()
    }

    component = render(<ProductSelectionBox {...props} />)
  })

  it('should render 3 block component', () => {
    expect(component.find('.row').children.length).toBe(3)
  })
})
