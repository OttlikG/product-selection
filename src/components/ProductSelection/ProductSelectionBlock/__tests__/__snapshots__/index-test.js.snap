// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSelectionBox should render 3 block component 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xs-4"
    >
      <Sports
        data={
          Array [
            Object {
              "category": "Sports",
              "id": "aaa",
              "isSelected": true,
              "location": "LONDON",
              "product_name": "Chelsea TV",
            },
            Object {
              "category": "Sports",
              "id": "bbb",
              "isSelected": false,
              "location": "LONDON",
              "product_name": "Arsenal TV",
            },
          ]
        }
        selectProductById={[Function]}
      />
    </div>
    <div
      className="col-xs-4"
    >
      <News
        data={Array []}
        selectProductById={[Function]}
      />
    </div>
    <div
      className="col-xs-4"
    >
      <Basket
        data={
          Array [
            Object {
              "category": "Sports",
              "id": "aaa",
              "isSelected": true,
              "location": "LONDON",
              "product_name": "Chelsea TV",
            },
          ]
        }
        push={[Function]}
        selectProductById={[Function]}
      />
    </div>
  </div>
</div>
`;
